security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -209,40 +237,27 @@ func GetSecretName(saName string) string {
 
 // Determine if the object is "enabled" for Istio.
 // Currently this looks at the list of watched namespaces and the object's namespace annotation
-func (sc *SecretController) istioEnabledObject(obj metav1.Object) bool {
-	if _, watched := sc.namespaces[obj.GetNamespace()]; watched || !sc.explicitOptIn {
+func (sc *SecretController) citadelManagedObject(obj metav1.Object) bool {
+
+	// todo(incfly)
+	// should be removed once listened namespaces flag phased out
+	if _, watched := sc.namespaces[obj.GetNamespace()]; watched {
 		return true
 	}
 
-	const label = "istio-managed"
-	enabled := !sc.explicitOptIn // for backward compatibility, Citadel always creates secrets
-	// @todo this should be changed to false once we communicate behavior change and ensure customers
-	// correctly mark their namespaces. Currently controlled via command line
-
 	ns, err := sc.core.Namespaces().Get(obj.GetNamespace(), metav1.GetOptions{})
-	if err != nil || ns == nil { // @todo handle errors? Unit tests mocks don't create NS, only secrets
-		return enabled
+	if err != nil || ns == nil { // if we can't retrieve namespace details, fall back on default value
```

Comment:
If you can't retrieve the namespace details, I think you should not consider the namespace is "enabled" to be safe. WDYT?
@incfly

Reply:
AFAIK, before this PR we don't check namespace label details, thus always 100% to proceed with handling secrets/key generation.

Fallback to default behavior seems align with existing behavior better.